plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostingFun/Core.hs

```diff
@@ -383,22 +392,53 @@ evaluateTwoVariableQuadraticFunction
   -- here: see Note [Minimum values for two-variable quadratic costing functions]
 {-# INLINE evaluateTwoVariableQuadraticFunction #-}
 
--- FIXME: we could use ModelConstantOrOneArgument for
--- ModelTwoArgumentsSubtractedSizes instead, but that would change the order of
--- the cost model parameters since the minimum value would come first instead of
--- last.
--- Tracked by https://github.com/IntersectMBO/plutus-private/issues/1554.
+-- | c00 + c01x*y + c12x*y^2
+-- This is used only for `expModInteger`, whose costing is quite complex.
+data ExpModCostingFunction = ExpModCostingFunction
+  { coefficient00 :: Coefficient00
+  , coefficient11 :: Coefficient11
+  , coefficient12 :: Coefficient12
+  } deriving stock (Show, Eq, Generic, Lift)
+  deriving anyclass (NFData)
+
+{- | Calculate the cost of calling `expModInteger a e m` where a is of size aa, e
+is of size ee, and m is of size mm.  If aa>5*mm then the cost is increased by
+50% to impose a penalty for the extra cost of initially reducing `a` modulo `m`.
+If large values of `a` really are required then the penalty can be avoided by
+calling `modInteger` before `expModInteger`.
+-}
+evaluateExpModCostingFunction
+  :: ExpModCostingFunction
+  -> CostingInteger
+  -> CostingInteger
+  -> CostingInteger
+  -> CostingInteger
+evaluateExpModCostingFunction
+   (ExpModCostingFunction
+    (Coefficient00 c00) (Coefficient11 c11) (Coefficient12 c12))
+  aa ee mm = if aa <= 5*mm
+          then cost0
+          else cost0 + (cost0 `dividedBy` 2)
```

Comment:
ie, 1.5 times the cost.  It seems a bit risky to multiply saturated integers by fractions though.  

I'm a bit worried about this being hard-coded since it'll have to be reporduced in eg the metatheory.  Maybe this should go in the cost model parameters along with the cofficients.